website/snippets/_render-method.md

```diff
@@ -0,0 +1,17 @@
+#### The render method
+
+The `.render()` method is generally used to resolve or evaluate Jinja expressions (such as `{{ source(...) }}`) during runtime. When a pre-hook or post-hook contains a dynamic reference, such as a table or column, dbt might not automatically resolve the reference correctly, particularly when certain flags (such as `--empty`) are applied.
```

Comment:
sugg removing the first paragraph to simplify the explanation since the 2nd paragraph already covers the key point about using .render() with the --empty flag